### ISSUE TITLE
Don't drain microtask queue for dynamic imports, as it breaks side-effect ordering

### ISSUE THREAD

**Issue by jhmaster2000** [oven-sh/bun#3191]
### What version of Bun is running?

0.6.7

### What platform is your computer?

Linux 5.15.90.1-microsoft-standard-WSL2 x86_64 x86_64

### What steps can reproduce the bug?

With `discord.js@14.11.0` and `express@4.18.2` installed, given the following files:
* `index.mjs`
```js
import('./bot.mjs');
import('./webserver.mjs');
```
* `bot.mjs`
```js
import { Client } from 'discord.js';

const client = new Client({ intents: 0 });

console.log('Logging in...');
try {
    // Valid token not needed for repro, leave this as is.
    await client.login('BADTOKEN');
} catch {} finally {
    console.log('Logged in or errored.');
}
```
* `webserver.mjs`
```js
import express from 'express';
const app = express();
app.get('/', (req, res) => { res.status(200).send('Webserver is online.'); });
app.listen(5080); // Change port if needed, not relevant for the repro.
console.info('Webserver is listening on port 5080.');
```

Run on node with `node index.mjs` and on bun with `bun index.mjs`

### What is the expected behavior?

The express webserver and the discord.js bot start/login in parallel arbitrary order, with the webserver successfully listening on the given port, and the discord.js bot completing the login() function (regardless of if successful or throwing an error), as displayed by the node output:
```
Webserver is listening on port 5080.
Logging in...
Logged in or errored.
```
or, because the parallel imports order is an arbitrary race condition it can also be, in any other arrangement of those 3 logs, but what matters is all 3 got printed.

### What do you see instead?

When running on Bun the output is only:
```
Webserver is listening on port 5080.
Logging in...
```
Notice the `Logged in or errored.` line never prints and bun hangs forever on the `await client.login(...)` call.

### Additional information

If the order of the imports in `index.mjs` is swapped around, as such:
```js
import('./webserver.mjs');
import('./bot.mjs');
```
There is no difference for node (works the same as usual), and it fixes the issue in bun as well, but obviously not a reliable or viable  workaround, especially for projects where there is more than just two parallel imports.
